src/Microsoft.Tye.Core/PublishProjectStep.cs

```diff
@@ -51,7 +51,7 @@ public override async Task ExecuteAsync(OutputContext output, ApplicationBuilder
                 throw new CommandException("'dotnet publish' failed.");
             }
 
-            output.WriteDebugLine($"Created Publish Output: '{outputDirectory}'");
+            output.WriteDebugLine($"Created Publish Output: '{outputDirectory.DirectoryPath}'");
```

Comment:
Found while trying to repro another issue. This was just printing the class name, `TempDirectory`.